fix: networktransform SwitchTransformSpaceWhenParented forces updates when InLocalSpace is true (#3664)
MidLevel/MLAPI: com.unity.netcode.gameobjects/Documentation~/components/core/networkobject.md

Reviewer: I think this paragraph changed as well. There is a new ownership overview page that this section should link to.

Reviewer: I would maybe ensure that the new ownership overview page is linked here. That way we don't need to try explain ownership in the note underneath.

```suggestion
To spawn a `NetworkObject` that is [owned](../../terms-concepts/ownership.md) by a different game client than the one doing the spawning, use the following:
```

```csharp
GetComponent<NetworkObject>().SpawnWithOwnership(clientId);
```
> [!NOTE]

> > When using `SpawnWithOwnership` method in a client-server network topology you must be aware that during the spawn sequence any component that has owner specific checks to perform specific actions will not be invoked on the server side. This can impact things like [NetworkTransform](../helper/networktransform.md) when set to an owner authority motion model. To avoid issues pertaining to this it is recommended to use `Spawn` where the server is the owner during the spawn sequence and then immediately following that with a call to `ChangeOwnership`. The two actions will get combined into a single `CreateObjectMessage` and will avoid potential, latency driven, issues.

Reviewer 2: ```suggestion
> When using the `SpawnWithOwnership` method in a client-server network topology, be aware that any component that has owner-specific checks to perform specific actions will not be invoked on the server side during the spawn sequence. This can impact things like [NetworkTransform](../helper/networktransform.md) when set to an owner authority motion model. To avoid issues, it's recommended to use `Spawn` where the server is the owner during the spawn sequence and then immediately follow with a call to `ChangeOwnership`. These two actions will get combined into a single `CreateObjectMessage` and avoid potential latency-driven issues.
```

```csharp
GetComponent<NetworkObject>().SpawnWithOwnership(clientId);
```
> [!NOTE]

> > When using the `SpawnWithOwnership` method, be aware that any component that has owner-specific checks to perform specific actions won't be invoked on the spawn authority  side during the spawn sequence. The spawn authority is the server when using a client-server network topology, and can be any client when using a distributed authority network topology. Using `SpawnWithOwnership` can impact things like [NetworkTransform](../helper/networktransform.md) when using an owner authority motion model, and potentially provide undesired parenting artifacts and/or impact your own scripts if you are planning to have the spawn authority make any further post-spawn adjustments within the same frame.

Reviewer: If we link the ownership overview above, I think we can simplify this wording some.

```suggestion
> The `SpawnWithOwnership` method allows a game client to spawn an object that it doesn't own. This means any owner-specific checks during the spawn sequence will not be invoked on the spawn authority side. Using `SpawnWithOwnership` can result in unexpected behavior if the spawn authority makes any additional post-spawn adjustments within the same frame.
```

(though looking at what's in that ownership page, I should expand the info about the motion model authority on that page)

> > To avoid potential issues, it's recommended to use `Spawn`, where the spawn authority starts as the owner throughout the spawn sequence, makes adjustments post-spawn, and then immediately follow with a call to `ChangeOwnership`.

Reviewer: Further re-wording

```suggestion
> To avoid potential issues, it's recommended to use `Spawn` if the spawn authority needs to make any adjustments post-spawn. After adjusting, the spawn authority can immediately follow with a call to `ChangeOwnership`.
```

```csharp
GetComponent<NetworkObject>().RemoveOwnership();
```
> [!NOTE]

> > It is not recommended to use `RemoveOwnership` when using a distributed authority network topology.

Reviewer 2: ```suggestion
> Using `RemoveOwnership` in a distributed authority network topology isn't recommended.
```
This might be obvious from context, but why isn't it recommended?

```csharp
GetComponent<NetworkObject>().RemoveOwnership();
```
> [!NOTE]

> > Using `RemoveOwnership` in a distributed authority network topology isn't recommended.

Reviewer: We actually throw an error in `RemoveOwnership` with DA ([NetworkSpawnManager.cs:L427](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/blob/d8f96d65ab90471893226766401b46059dd054f6/com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs#L427))

```suggestion
> `RemoveOwnership` isn't supported when using a [distributed authority network topology](../../terms-concepts/distributed-authority.md).
```
